include/psen_scan_v2/scan_range.h

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2020 Pilz GmbH & Co. KG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#ifndef PSEN_SCAN_V2_SCAN_RANGE_H
+#define PSEN_SCAN_V2_SCAN_RANGE_H
+
+#include <stdexcept>
+
+#include "psen_scan_v2/tenth_of_degree.h"
+
+namespace psen_scan_v2
+{
+/**
+ * @brief Higher level data type storing the range in which the scanner takes measurements.
+ */
+template <uint16_t min_allowed_angle, uint16_t max_allowed_angle>
```

Comment:
I am not that comfortable with values as template parameters since for every set of template arguments a class is compiled this does not scale well. Also you run into issues if you want to implement operators etc... see https://godbolt.org/z/1jf5vj

In this case at hand you also don't check for min_allowed_angle <= max_allowed_angle

Maybe ScanRange could have an (optional?) limits parameter of type ScanRange? Or there is LimitedScanRange inheriting from ScanRange?

Reply:
This would also not work if the angles would have to be double.